apps/mark-scan/frontend/src/app_root.tsx

```diff
@@ -457,10 +457,24 @@ export function AppRoot({
     }
 
     if (
-      isPollWorkerAuth(authStatus) &&
-      // Ballot invalidation requires BallotContext, handled below
-      stateMachineState !== 'waiting_for_invalidated_ballot_confirmation'
+      (isPollWorkerAuth(authStatus) || isCardlessVoterAuth(authStatus)) &&
+      (stateMachineState ===
+        'waiting_for_invalidated_ballot_confirmation.paper_present' ||
+        stateMachineState ===
+          'waiting_for_invalidated_ballot_confirmation.paper_nowhere')
     ) {
+      return (
+        <BallotInvalidatedPage
+          authStatus={authStatus}
+          paperPresent={
+            stateMachineState ===
+            'waiting_for_invalidated_ballot_confirmation.paper_present'
+          }
```

Comment:
Seemed clearer to pass this prop down rather than query for state in the lower components, but open to feedback

Reply:
seems fine to me!